src/wasm-interpreter.h

```diff
@@ -1454,14 +1454,14 @@ class ExpressionRunner : public OverriddenVisitor<SubType, Flow> {
       auto* func = module->getFunction(cast.originalRef.getFunc());
       seenRtt = Literal(Type(Rtt(0, func->sig)));
       cast.castRef =
-        Literal(func->name, Type(intendedRtt.type.getHeapType(), Nullable));
+        Literal(func->name, Type(intendedRtt.type.getHeapType(), NonNullable));
```

Comment:
Interesting question. We know this is not a null (line 1435), so it is valid to mark it as either nullable or not, I think. This is just a runtime value, and not the type of the expression. So even if the expression has a nullable type, it's fine to return a non-nullable runtime value (since it's always ok to return a subtype).

I can't think of a reason why this would matter in practice. I think the more specific type is clearer?

Reply:
Yeah I think that shouldn't hurt.